moveit_kinematics/ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.sh

```diff
@@ -0,0 +1,212 @@
+	else
+		ROBOT_NAME="$robot_name"
+	fi
+}
+
+function create_dae_file {
+	echo "Converting urdf to Collada"
+	if ! rosrun collada_urdf urdf_to_collada "$INPUT" "$DAE_FILE" 2> /dev/null ; then
+		# When this failed, run docker
+		build_docker_image
+		echo "Converting urdf to Collada"
+		cp "$INPUT" "$TMP_DIR/robot.urdf"
+		test "$QUIET" == "1" && exec 3>&2 2>/dev/null  # save stderr as 3, then redirect to /dev/null
+		docker run --rm --user $(id -u):$(id -g) -v $TMP_DIR:/input --workdir /input -e HOME=/input \
+				 fixed-openrave:latest rosrun collada_urdf urdf_to_collada robot.urdf robot.dae
+		test "$QUIET" == "1" && exec 2>&3  # restore stderr
+	fi
+}
+
+function create_solver {
+	# create wrapper.xml file, needed by OpenRave
+	cat <<EOF > $TMP_DIR/wrapper.xml
+<robot file="robot.dae">
+  <Manipulator name="$ROBOT_NAME">
+    <base>$BASE_LINK</base>
+    <effector>$EEF_LINK</effector>
+  </Manipulator>
+</robot>
+EOF
+
+	build_docker_image
+
+	# Assemble openrave command
+	cmd="openrave0.9.py --database inversekinematics --robot=/input/wrapper.xml --iktype=$IK_TYPE --iktests=1000"
+	echo "Running $cmd"
+
+	# save stdout as 3, then redirect to /dev/null
+	test "$QUIET" == "1" && exec 3>&1 1>/dev/null
+	# run $cmd in docker as current user, outputting files to $TMP_DIR/.openrave
+	docker run --rm --user $(id -u):$(id -g) \
+		-v $TMP_DIR:/input --workdir /input -e HOME=/input \
+		fixed-openrave:latest $cmd
+
+	# restore stdout
+	test "$QUIET" == "1" && exec 1>&3
+
+	# update INPUT to generated .cpp
+	INPUT=$(ls -1 $TMP_DIR/.openrave/*/*.cpp 2> /dev/null)
+	if [ -n "$INPUT" ] ; then
+		echo "Created $INPUT"
+	else
+		echo "Failed to create ikfast solver"
+		exit 1
+	fi
+}
+
+function create_plugin {
+	echo
+	echo "Running $(dirname $0)/create_ikfast_moveit_plugin.py"
+	$(dirname "$0")/create_ikfast_moveit_plugin.py "$ROBOT_NAME" "$PLANNING_GROUP" "$PKG_NAME" "$BASE_LINK" "$EEF_LINK" "$INPUT"
+}
+
+### main program ###
+
+parse_options $*
+
+# Create a temporary directory to operate in
+TMP_DIR=$(mktemp -d --tmpdir ikfast.XXXXXX)
+DAE_FILE="$TMP_DIR/robot.dae"
+if [ "$KEEP_RESULTS" == "1" ] ; then
+	echo "Storing intermediate results in: $TMP_DIR"
+else
+	# Register cleanup function to be called on EXIT signal
+	trap cleanup EXIT
+fi
+
+# Depending on input file type, select the required processing stages
+while true ; do
+	filename=$(basename -- "$INPUT")
+	extension="${filename##*.}"
+	case "$extension" in
+		urdf|URDF)  # create .dae from .urdf
+			extract_robot_name "$INPUT"
+			create_dae_file
+			INPUT="$DAE_FILE"
+			;;
+		dae|DAE)  # create solver .cpp
+			if ! [ "$INPUT" -ef "$DAE_FILE" ] ; then
+				cp "$INPUT" "$DAE_FILE"
+			fi
+			set_option_defaults
+			create_solver
+			;;
+		cpp)  # create wrapper plugin
+			set_option_defaults
+			create_plugin
+			break
+			;;
+		*)
+			echo "Unknown input file type '.$extension'. Expecting .urdf, .dae, or .cpp."
+			exit 1
+			;;
+	esac
+done
```

Comment:
overall this is a pretty awesome script!

Reply:
I was actually thinking to merge this into the .py script as well. But I don't have more cycles for this now.